### DISCUSSION TITLE
Why Java 17?

### DISCUSSION THREAD

**Discussion by Phillipus** [eclipse-m2e/m2e-core#891] [Q&A] [answered]
m2e latest versions require Java 17. Can I ask why when other Eclipse projects still support Java 11 and even 8?

**Comment by mickaelistria** [comment, marked as answer]
Java 17 leads to higher quality code (some bugs can easily be avoided with newer constructs or APIs that leave less rooms for mistakes) and also to better performance. m2e project benefits from it.
And some other project m2e depends on (for example TM4E) do already require Java 17, and Eclipse SimRel has allowed dependencies on Java 17 for 2 releases now.

**Reply by Phillipus** [reply]
Thanks for the info. I was just curious.

**Comment by HannesWell** [comment]
Besides the already mentioned reasons the 2.0 release, which was necessary due to API removals, was also a good opportunity to lift the minimum Java-version. Increasing the minimum Java-version is often considered a major API break and therefore 2.0 was a good fit. Otherwise we should probably have done a major release in the not too far future when the rest of the Eclipse-Platform is moving to Java-17. I don't know when this will happen, but I guess it will be within the next few years (or maybe even less).
